packages/graphile-utils/src/makeChangeNullabilityPlugin.ts

```diff
@@ -1,49 +1,121 @@
+      type: Type,
+      allNullabilities: boolean[]
+    ): Type {
+      const [nullability, ...rest] = allNullabilities;
+      const nullableType = type instanceof GraphQLNonNull ? type.ofType : type;
+      let inner = null;
+      if (nullableType instanceof GraphQLList) {
+        inner = new GraphQLList(applyNullability(nullableType.ofType, rest));
+      } else {
+        if (rest.length > 0) {
+          throw new Error(`Too many booleans passed`);
+        }
+        inner = nullableType;
+      }
+      return nullability
+        ? inner
+        : inner === type
+        ? new GraphQLNonNull(inner)
+        : type; // Optimisation if it's already non-null*/;
+    }
+    function applyTypeNullability<Type extends GraphQLType | GraphQLInputType>(
+      type: Type,
+      shouldBeNullable: boolean | boolean[]
+    ): Type {
+      const allNullabilities: boolean[] =
+        shouldBeNullable instanceof Array
+          ? shouldBeNullable
+          : [shouldBeNullable];
+      return allNullabilities.length > 0
+        ? applyNullability(type, allNullabilities)
+        : type;
+    }
     function changeNullability<
       Field extends GraphQLInputFieldConfig | GraphQLFieldConfig<any, any>
     >(field: Field, build: Build, context: Context<Field>): typeof field {
       const {
         Self,
         scope: { fieldName },
       } = context;
-      const typeRules = rules[Self.name];
+      const typeName = Self.name;
+      const typeRules = rules[typeName];
       if (!typeRules) {
         return field;
       }
       const shouldBeNullable = typeRules[fieldName];
       if (shouldBeNullable == null) {
         return field;
       }
-      const {
-        graphql: { getNullableType, GraphQLNonNull },
-      } = build;
-      const nullableType = getNullableType(field.type);
-      return {
-        ...field,
-        type: shouldBeNullable
-          ? nullableType
-          : nullableType === field.type
-          ? new GraphQLNonNull(field.type)
-          : field.type, // Optimisation if it's already non-null
-      };
+      const shouldApplyNullabilities: NullabilityOpts = {};
+      if (
+        typeof shouldBeNullable === "object" &&
+        !(shouldBeNullable instanceof Array)
+      ) {
+        shouldApplyNullabilities.type = shouldBeNullable.type;
+        shouldApplyNullabilities.args = shouldBeNullable.args;
+      } else {
+        shouldApplyNullabilities.type = shouldBeNullable;
+      }
+      const updatedField = { ...field };
+      try {
+        if (shouldApplyNullabilities.type) {
+          updatedField.type = applyTypeNullability(
+            field.type,
+            shouldApplyNullabilities.type
+          );
+        }
+        if (shouldApplyNullabilities.args) {
+          Object.entries(shouldApplyNullabilities.args).forEach(
+            ([argName, argShouldBeNullable]) => {
+              if (!("args" in field) || !field.args?.[argName]) return;
```

Comment:
This line should probably throw, or at least log a warning.

Reply:
✅